### PR TITLE
T20 - Display the individual digits of a given number

### PR DESCRIPTION
**Pull request by karthick-1205**

### CODE REVIEW

File: MainRepository/Program.cs [hunk 1]
```diff
@@ -1,7 +1,29 @@
...
+            string resultString = String.Join(" ", int_part_array);
+            string resultString1 = String.Join(" ", fract_part_array);
+            if (resultString.Length > 0 && resultString1.Length == 0) Console.Write($"Integral part:{resultString}\n");
+            if (resultString1.Length > 0 && resultString.Length == 0) Console.Write($"Factorial part:{resultString1}\n");
+            if (resultString.Length > 0 && resultString1.Length > 0)
+            {
+                Console.Write($"Integral part:{resultString} ");
+                Console.WriteLine($"Factorial part:{resultString1}");
+            }
+
```
Comment (priyarathna) [inline MainRepository/Program.cs]: Remove the empty lines.

File: MainRepository/Program.cs [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
+    Console.Write("Enter the number or press [e] to exit: ");
+    string input = Console.ReadLine();
+    if (input.ToLower() == "e") break;
+    if (double.TryParse(input, out double n) && n > 0)
     {
-        Console.WriteLine("Hello, World!");
+        if (input.Contains("."))
+        {
+            string[] result = input.Split('.');
+            char[] int_part_array = result[0].ToCharArray();
```
Comment (priyarathna) [inline MainRepository/Program.cs]: Rename: int_part_array => int_arr, fract_part_array=> fract_arr

File: MainRepository/Program.cs [hunk 3]
```diff
@@ -1,7 +1,29 @@
...
+            char[] fract_part_array = result[1].ToCharArray();
+            string resultString = String.Join(" ", int_part_array);
+            string resultString1 = String.Join(" ", fract_part_array);
+            if (resultString.Length > 0 && resultString1.Length == 0) Console.Write($"Integral part:{resultString}\n");
```
Comment (priyarathna) [inline MainRepository/Program.cs]: I couldn't understand the use of the following:
if (resultString.Length > 0 && resultString1.Length == 0) Console.Write ($"Integral part:{resultString}\n");
if (resultString1.Length > 0 && resultString.Length == 0) Console.Write ($"Factorial part:{resultString1}\n");
as you're handling only doubles. If I give 56 as input, it is not a valid input.

File: MainRepository/Program.cs [hunk 4]
```diff
@@ -1,7 +1,29 @@
...
+for (; ; )
 {
-    private static void Main(string[] args)
+    Console.Write("Enter the number or press [e] to exit: ");
+    string input = Console.ReadLine();
+    if (input.ToLower() == "e") break;
+    if (double.TryParse(input, out double n) && n > 0)
     {
-        Console.WriteLine("Hello, World!");
+        if (input.Contains("."))
```
Comment (priyarathna) [inline MainRepository/Program.cs]: No need of this as you're using double.TryParse. The input "4." is invalid.

File: MainRepository/Program.cs [hunk 5]
```diff
@@ -1,7 +1,21 @@
-﻿internal class Program
+﻿//Display the individual digits of given number
+while (true)
```
Comment (priyarathna) [inline MainRepository/Program.cs]: Is there a reason why you change for loop to while loop?

Comment (karthick-1205) [reply MainRepository/Program.cs]: In most assignments, I used infinite for loop, so I changed to infinite while loop for practice.

File: MainRepository/Program.cs (hunk 4 above)
Comment (priyarathna) [reply MainRepository/Program.cs]: @karthick-1205 : This is not yet fixed right?

File: MainRepository/Program.cs (hunk 1 above)
Comment (priyarathna) [reply MainRepository/Program.cs]: @karthick-1205 : Are the empty lines has been removed?

File: MainRepository/Program.cs [hunk 6]
```diff
@@ -1,21 +1,20 @@
...
+while (true) {
+   Console.Write ("Enter the number or press [e] to exit: ");
+   string input = Console.ReadLine ();
+   if (input.ToLower () == "e") break;
+   if (double.TryParse (input, out double d) && d > 0) {
+      if (input.Contains (".")) {
+         string[] split = input.Split ('.');
+         string integral = string.Join (" ", split[0].ToCharArray ());
+         string factorial = string.Join (" ", split[1].ToCharArray ());
+         if (integral.Length > 0 && factorial.Length == 0) {
```
Comment (priyarathna) [inline MainRepository/Program.cs]: No need of flower-brace if it's a single statement.

**Review by priyarathna — APPROVED** [review]

**Pull request closed without merging**